src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -288,6 +272,105 @@ void MemoryMngrRealloc::destroy(void *ptr) {
     dnnl::impl::free(ptr);
 }
 
+/////////////// StringMemory ///////////////
+
+StringMemory::StringMemory(const dnnl::engine& engine, const MemoryDescPtr& desc, const OvString* data) : m_engine(engine), m_mem_desc(desc) {
+    m_manager = std::make_shared<StringMemoryMngr>();
+
+    if (!m_mem_desc->isDefined()) {
+        return;
+    }
+
+    m_size = m_mem_desc->getCurrentMemSize();
+    const auto string_size = m_mem_desc->getShape().getElementsCount();
+
+    if (data != nullptr) {
+        m_manager->setExtStringBuff(const_cast<OvString *>(data), string_size);
+    } else {
+        m_manager->resize(string_size);
+    }
+}
+
+StringMemory::OvString* StringMemory::StringMemoryMngr::getStringPtr() const noexcept {
+    return m_data.get();
+}
+
+void StringMemory::load(const IMemory& src, bool ftz) const {
+    transferData(src, *this, false);
+}
+
+void* StringMemory::getData() const  {
+    return m_manager->getRawPtr();
+}
+
+void StringMemory::redefineDesc(MemoryDescPtr desc) {
```

Comment:
Please add precision check, only `string ` is allowed.

Reply:
Done